purchase_discount/views/purchase_discount_view.xml

```diff
@@ -29,13 +29,19 @@
                 expr="//field[@name='order_line']/tree/field[@name='price_unit']"
                 position="after"
             >
-                <field name="discount" />
+                <field
+                    name="discount"
+                    attrs="{'readonly': [('parent.invoice_status', '=', 'invoiced')]}"
```

Comment:
Why don't use same condition that price_unit:
https://github.com/odoo/odoo/blob/51b693fa294ac3ee28cf55164bd14afa0d92a216/addons/purchase/views/purchase_views.xml#L252


```suggestion
                    attrs="{'readonly': [('qty_invoiced', '!=', 0)]}"
```